enums.rb

```diff
@@ -0,0 +1,128 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # rubocop:enable Metrics/ModuleLength
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  def my_all?(arg = nil)
+    if block_given? && !arg.nil?
+      return false unless my_all?(arg) && my_all?
+    end
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+    elsif arg.nil?
+      my_each { |i| return false unless i }
+    elsif arg.class == Class
+      my_each { |i| return false unless i.class <= arg }
+    elsif arg.class == Regexp
+      my_each { |i| return false unless i =~ arg }
+    else
+      my_each { |i| return false unless i == arg && i.class <= arg.class }
+    end
+
+    true
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |i|
+      new_array.push(i) if yield(i)
+    end
+    new_array
+  end
+
+  def my_any?(arg = nil)
+    if block_given? && !arg.nil?
+      return true unless my_any?(arg) && my_any?
+    end
+
+    if block_given?
+      my_each { |i| return true unless yield(i) }
+    elsif arg.nil?
+      my_each { |i| return true unless i }
+    elsif arg.class == Class
+      my_each { |i| return true unless i.class <= arg }
+    elsif arg.class == Regexp
+      my_each { |i| return true unless i =~ arg }
+    else
+      my_each { |i| return true unless i == arg && i.class <= arg.class }
+    end
+
+    false
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

```ruby     
[nil, false].my_any? == [nil, false].any? => false
```
- Yours returns true (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/86542092-82315400-becf-11ea-8e94-2325f63e9017.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/86542096-8f4e4300-becf-11ea-8f02-1752f508209e.png)

```ruby     
[5, -1, true].my_any? == [5, -1, true].any? => true
```
- Yours returns false (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/86542092-82315400-becf-11ea-8e94-2325f63e9017.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/86542213-ec96c400-bed0-11ea-81f9-d03d2e4f8200.png)

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

```ruby
['1', '2.5', 'a', '9'].my_any?(Integer) == ['1', '2.5', 'a', '9'].any?(Integer) => false
```
- Yours is retrieving true (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/86542271-91b19c80-bed1-11ea-80a2-48a800654156.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/86542284-a2621280-bed1-11ea-9347-2d14ea633fbc.png)

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
%w[dog door rod blade].my_any?(/z/) == %w[dog door rod blade].any?(/z/) => false
```
- Yours retrieves true (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/86542296-cd4c6680-bed1-11ea-846e-329d7f42a56f.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689742-cf646800-6f67-11ea-9e3e-eec7a960fe38.png)

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

```ruby
[4,7,11].my_any?(3) should be equal to this [4,7,11].any?(3) => false
```
- Yours retrieves true (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/86542954-7c3f7100-bed7-11ea-9f95-bc8486945a90.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/86542963-88c3c980-bed7-11ea-8f97-730657ccb215.png)